jpsonic-main/src/main/java/com/tesshu/jpsonic/service/scanner/ScannerStateServiceImpl.java

```diff
@@ -27,9 +32,18 @@ public class ScannerStateServiceImpl implements ScannerStateService {
 
     private final AtomicBoolean cleansing = new AtomicBoolean(true);
 
-    public ScannerStateServiceImpl(IndexManager indexManager) {
+    /**
+     * Scan start time. Use only within the thread that acquired the lock.
+     */
+    private Instant scanDate = FAR_PAST;
+
+    public ScannerStateServiceImpl(StaticsDao staticsDao) {
         super();
-        this.indexManager = indexManager;
+        this.staticsDao = staticsDao;
+    }
+
+    Instant getScanDate() {
+        return scanDate;
```

Comment:
@sonatype-lift ignore

1. tryScanningLock() (=true)
2.getScanDate()
3. unlockScanning()

executed in this order.
Also, since this can only be done in one thread, writing to private fields will not race.

Reply:
@sonatype-lift ignore